src/fparser/two/utils.py

```diff
@@ -996,8 +996,15 @@ def match(lhs_cls, rhs_cls, string, require_lhs=True, upper_lhs=False):
         # an '=', e.g. FMT='("Hello = False")'
         from fparser.two.Fortran2003 import Char_Literal_Constant
         if not Char_Literal_Constant.match(string) and "=" in string:
+            # Only split on the left-most '=' character
             lhs, rhs = string.split('=', 1)
             lhs = lhs.rstrip()
+            if not pattern.name.match(lhs.lstrip()):
```

Comment:
I have to say I find the logic of this class a bit strange and I'm not sure how robust it is. There seems to be an assumption that the only way an invalid `=` would appear is when the whole string is a Char_Literal_Constant, but then you find that this is not always the case so add an additional check that the lhs is a name.

If we are going for minimal changes then I think you should match against `lhs_cls` not name.

If we are going for an improvement then I think we can make the assumption that the lhs will never have an `=` in it - but would have to document that in the docstring. This seems reasonable seeing as it is meant to be a keyword. If so, we can simply split on the leftmost `=` if there is one and check that the lhs of the split is a valid keyword. If so, fine, if not try no lhs and all rhs.